ui/packages/components/src/utils/useDevServer.tsx

```diff
@@ -45,13 +45,30 @@ async function sendToDevServer(payload: string) {
   }
 }
 
-export function useDevServer() {
+export function useDevServer(pollingInterval?: number) {
   const [isRunning, setRunning] = useState<boolean>(false);
+
   useEffect(() => {
-    fetch(devServerURL)
-      .then(() => setRunning(true))
-      .catch(() => setRunning(false));
-  }, []);
+    const checkServerStatus = () => {
+      fetch(devServerURL)
+        .then(() => setRunning(true))
+        .catch(() => setRunning(false));
+    };
+
+    checkServerStatus();
+
+    let intervalId: NodeJS.Timeout | undefined;
```

Comment:
Hrm...since useEffect will only ever run in the browser, using node types is probably not what we want. 
^ not a blocker since it's just type meta.